### ISSUE TITLE
Robot freeze util I give a little push to the camera in hand

### ISSUE THREAD

**Issue by zhimou** [IliaLarchenko/dot_policy#4]
Hi Ilia,

Thanks for sharing your excellent work! I have learned a lot from your work. 

I have migrated your work to lerobot and I have trained it with 60 episodes' data to pick up a little doll to a box. I noticed that the robot do not move until I push the camera in hand a bit(or I move the pick up target a little bit also works). After this push, this policy works better than ACT and diffusion policy in terms of speed, accuracy and fluency. 

I wonder what causes this phenomenon and how to fix this issue?

Below is my setup, I have two cameras: one in hand and another camera look down to the table(just like what you advice in your video).

![Image](https://github.com/user-attachments/assets/a723f650-a1de-4027-a4a7-4d80215e9637)

That is how it looks like:

https://github.com/user-attachments/assets/105c108e-5c94-4a5b-9772-e3e49a88ba2a

Look forward to you reply!

**Comment by IliaLarchenko** [comment]
I guess it's happening because of the pauses in your training data - the same issue I show in my video, where the robot gets stuck right before dropping the ball into the cup.

When you record your dataset, you probably have a short delay before moving the arm, and the policy learns to do the same, getting stuck in the initial position.

Let’s take an example: say your FPS = 30, and you predict 1s forward (i.e., 30 actions per chunk). Every time you record data, you start moving the robot only after 3 seconds (90 steps) from the beginning of the recording.

In that case, the first 60 datapoints of each episode will be identical: the robot is in the initial state, and the target (next 30 actions) is just to stay in that state. Only 30 datapoints (starting from frame 61+) will have any meaningful actions - at least some difference in the next 30 steps.

Eventually, the policy learns that if it’s in the initial state and not moving, it should stay in that state and keep still. When you push it a bit, it is not in the initial fixed position anymore, and it starts moving.

Diffusion policy doesn’t have this problem because it introduces some randomness in predictions and eventually kicks itself out of the stuck state.

ACT can theoretically have the same issue - it’s partly a matter of luck whether the model learns to avoid it.

One solution is to extend the prediction horizon: increase `inference_horizon` and `train_horizon`, and optionally increase `alpha` and `train_alpha`.

If `inference_horizon < train_horizon` and `alpha < 1.0`, try running the trained model with `inference_horizon = train_horizon` and `alpha = 1.0`. It is the easiest step, and can help even without retraining.

If that doesn’t help, try retraining (or fine-tuning) the model using a higher horizons and alphas.

**Comment by zhimou** [comment]
> I guess it's happening because of the pauses in your training data - the same issue I show in my video, where the robot gets stuck right before dropping the ball into the cup.
> 
> When you record your dataset, you probably have a short delay before moving the arm, and the policy learns to do the same, getting stuck in the initial position.
> 
> Let’s take an example: say your FPS = 30, and you predict 1s forward (i.e., 30 actions per chunk). Every time you record data, you start moving the robot only after 3 seconds (90 steps) from the beginning of the recording.
> 
> In that case, the first 60 datapoints of each episode will be identical: the robot is in the initial state, and the target (next 30 actions) is just to stay in that state. Only 30 datapoints (starting from frame 61+) will have any meaningful actions - at least some difference in the next 30 steps.
> 
> Eventually, the policy learns that if it’s in the initial state and not moving, it should stay in that state and keep still. When you push it a bit, it is not in the initial fixed position anymore, and it starts moving.
> 
> Diffusion policy doesn’t have this problem because it introduces some randomness in predictions and eventually kicks itself out of the stuck state.
> 
> ACT can theoretically have the same issue - it’s partly a matter of luck whether the model learns to avoid it.
> 
> One solution is to extend the prediction horizon: increase `inference_horizon` and `train_horizon`, and optionally increase `alpha` and `train_alpha`.
> 
> If `inference_horizon < train_horizon` and `alpha < 1.0`, try running the trained model with `inference_horizon = train_horizon` and `alpha = 1.0`. It is the easiest step, and can help even without retraining.
> 
> If that doesn’t help, try retraining (or fine-tuning) the model using a higher horizons and alphas.

You are right! Most of my data episodes have more than 2 second's pause at the beginning and my training and inference horizon is 20, and my actual fps is above 10 hz, hence model learn to pause at the beginning(80% of time it pause).
I tried changing alpha without retraining, but without any luck. Then I set train_horizon and inference_horizon to 100(100/10fps is much greater then 2 seconds). After retraining the model, it can pick up the doll without pushing it in about 80% of time. **But there is still about 20% chance that it freeze completely and need a little push to the wrist camera(or interfere the camera, both way like what you say, get robot out of initial state) to activate the model, and increasing alpha to 1 seems doesn't help.** I will try to tune the hyper parameters of the model and collect more data (with less pause at the beginning) to try further.
Thanks for your help!
